### ISSUE TITLE
IndexOutOfRangeException when pressing unknown key

### ISSUE THREAD

**Issue by Sparta142** [opentk/opentk#1047]
### Description

Pressing an "unknown" key causes an `IndexOutOfRangeException` when the key code is -1.

In my experience, all the media keys on my keyboard (previous, play/pause, next, mute, volume up, and volume down) count as unknown keys.

### Repro steps

1. Create new .NET Core project with the example code below
2. Install **OpenTK 4.0.0-pre8** from NuGet
3. Run the project and press a media key (e.g., volume up)

#### Example code
```c#
using OpenToolkit.Windowing.Desktop;

internal static class Program
{
    private static void Main()
    {
        using var window = new GameWindow(new GameWindowSettings(), new NativeWindowSettings());
        window.Run();
    }
}
```

### Expected behavior

No unhandled exception or crash. Some sort of graceful handling of unknown GLFW keys.

### Actual behavior

The library throws an exception, crashing the program.

### Related information 

Looks like the offending code is ~~[NativeWindow.cs:689](https://github.com/opentk/opentk/blob/master/src/OpenToolkit.Windowing.Desktop/NativeWindow.cs#L689)~~ in RegisterWindowCallbacks in NativeWindow.cs, where the key callback is set. I didn't check with a clone of the repo though.

#### Stack trace
```
System.IndexOutOfRangeException
  HResult=0x80131508
  Message=Index was outside the bounds of the array.
  Source=OpenToolkit.Windowing.Desktop
  StackTrace:
   at OpenToolkit.Windowing.Desktop.NativeWindow.<RegisterWindowCallbacks>b__139_5(Window* window, Keys key, Int32 scancode, InputAction action, KeyModifiers mods)
   at OpenToolkit.Windowing.GraphicsLibraryFramework.GLFWNative.glfwPollEvents()
   at OpenToolkit.Windowing.Desktop.NativeWindow.ProcessEvents()
   at OpenToolkit.Windowing.Desktop.GameWindow.Run()
   at Program.Main() in C:\Users\spart\source\repos\ConsoleApp1\Program.cs:line 8
```

#### Environment
* Windows 10 Pro 1909 x64
* GLFW backend
* OpenTK 4.0.0-pre8 (NuGet)
* .NET Core 3.1 (CoreCLR 3.1.3)

#### Workaround
* Replace the GLFW key callback from within your NativeWindow subclass with a callback that does a check for a -1 key code before indexing into GlfwKeyMapping.

**Comment by arakis** [comment]
I can confirm this issue on Debian / Linux.

**Comment by varon** [comment]
Fixed by #1051

**Comment by varon** [comment]
Thanks for the report!
